### PR TITLE
Add `serde` support

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -86,6 +86,8 @@ pub const H: u32 = 1 << 31;
 
 /// Child index, whether hardened or not
 #[derive(Clone, Copy, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize), serde(into = "u32"))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize), serde(from = "u32"))]
```
Comment (maurges) [inline src/lib.rs:90]:
```
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize), serde(into = "u32", from = "u32"))]
```
?

Comment (survived) [reply src/lib.rs:90]: Too long, it'll be formatted awfully

Comment (maurges) [reply src/lib.rs:90]: Formatted:
```
-#[cfg_attr(feature = "serde", derive(serde::Serialize), serde(into = "u32"))]
-#[cfg_attr(feature = "serde", derive(serde::Deserialize), serde(from = "u32"))]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(into = "u32", from = "u32")
+)]
```
I mean, I think I like it better this formatted way

Comment (survived) [reply src/lib.rs:90]: I don't, I like how it is now

Comment (survived) [reply src/lib.rs:90]: It even adds beauty as serialization-related stuff is in the first line, deserialization-related is in the second

**Review by maurges — APPROVED** [review]

**Pull request merged**